shapely/constructive.py

```diff
@@ -902,18 +914,63 @@ def snap(geometry, reference, tolerance, **kwargs):
 
     Examples
     --------
-    >>> from shapely import LineString, Point, Polygon
+    >>> from shapely import snap, distance, LineString, Point, Polygon, MultiPolygon, MultiPoint, box
+
     >>> point = Point(0.5, 2.5)
     >>> target_point = Point(0, 2)
     >>> snap(point, target_point, tolerance=1)
     <POINT (0 2)>
     >>> snap(point, target_point, tolerance=0.49)
     <POINT (0.5 2.5)>
+
     >>> polygon = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
     >>> snap(polygon, Point(8, 10), tolerance=5)
     <POLYGON ((0 0, 0 10, 8 10, 10 0, 0 0))>
     >>> snap(polygon, LineString([(8, 10), (8, 0)]), tolerance=5)
     <POLYGON ((0 0, 0 10, 8 10, 8 0, 0 0))>
+
+    # snapping a line to line (demonstrates cleaning imprecise coordinates)
```

Comment:
```suggestion
    You can snap one line to another, for example to clean imprecise coordinates:
    
```

You can just use text, so long as it is separated by newlines.  See [examples in `strtree.py`](https://github.com/shapely/shapely/blob/2.0.0/shapely/strtree.py#L168) for instance

Then the text blocks can be more descriptive of the behavior that follows (sorry, mine didn't do this well when I added them in review comments previously)

e.g.,

```
You can snap all 3 coordinates in 3D geometries:

>>> ...
```

```
Snapping to an empty geometry has no effect:

>>> ...
```

etc.